src/data/CHDatasource.ts

```diff
@@ -32,6 +57,129 @@ export class Datasource extends DataSourceWithBackend<CHQuery, CHConfig> {
     this.adHocFilter = new AdHocFilter();
   }
 
+  getDataProvider(
+    type: SupplementaryQueryType,
+    request: DataQueryRequest<CHQuery>
+  ): Observable<DataQueryResponse> | undefined {
+    if (!this.getSupportedSupplementaryQueryTypes().includes(type)) {
+      return undefined;
+    }
+    switch (type) {
+      case SupplementaryQueryType.LogsVolume:
+        const logsVolumeRequest = cloneDeep(request);
+
+        const timespanMs = logsVolumeRequest.range.to.valueOf() - logsVolumeRequest.range.from.valueOf();
+        const intervalInfo = getIntervalInfo(logsVolumeRequest.scopedVars, timespanMs);
+
+        logsVolumeRequest.interval = intervalInfo.interval;
+        logsVolumeRequest.scopedVars.__interval = { value: intervalInfo.interval, text: intervalInfo.interval };
+
+        if (intervalInfo.intervalMs !== undefined) {
+          logsVolumeRequest.intervalMs = intervalInfo.intervalMs;
+          logsVolumeRequest.scopedVars.__interval_ms = {
+            value: intervalInfo.intervalMs,
+            text: intervalInfo.intervalMs,
+          };
+        }
+
+        logsVolumeRequest.hideFromInspector = true;
```

Comment:
I moved this out of `queryLogsVolume` here, as I thought it would be a bit cleaner to mutate the request only in one place